wagtail/images/management/commands/wagtail_update_image_renditions.py

```diff
@@ -0,0 +1,75 @@
+from django.core.management.base import BaseCommand
+
+from wagtail.images import get_image_model
+
+
+class Command(BaseCommand):
+    """Command to create missing image renditions with the option to remove (purge) any existing ones."""
+
+    help = "This command will generate all image renditions, with an option to purge existing renditions first."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--purge",
+            action="store_true",
+            help="Purge and regenerate all image renditions",
+        )
+        parser.add_argument(
+            "--purge-only",
+            action="store_true",
+            help="Purge all image renditions without regenerating them",
+        )
+
+    def handle(self, *args, **options):
+        renditions = get_image_model().get_rendition_model().objects.all()
+        if len(renditions) == 0:
+            self.stdout.write("No image renditions found.")
+            return
+
+        success_count = 0
+        if options["purge"]:
+            for rendition in renditions:
+                try:
+                    rendition_filter = rendition.filter
+                    rendition_image = rendition.image
+                    rendition.delete()
+                    rendition_image.get_rendition(rendition_filter)
+                    success_count = success_count + 1
+                except Exception:
+                    self.stderr.write(
+                        f"Could not purge and regenerate rendition for {rendition_image.title}"
+                    )
+            self.stdout.write(
+                self.style.SUCCESS(
+                    f"Successfully purged and regenerated {success_count} image renditions!"
+                )
+            )
+        elif options["purge_only"]:
+            for rendition in renditions:
+                try:
+                    rendition_image = rendition.image
+                    rendition.delete()
+                    success_count = success_count + 1
+                except Exception:
+                    self.stderr.write(
+                        f"Could not purge rendition for {rendition_image.title}"
+                    )
+            self.stdout.write(
+                self.style.SUCCESS(
+                    f"Successfully purged {success_count} image renditions!"
+                )
+            )
+        else:
+            for rendition in renditions:
+                try:
+                    rendition.image.get_rendition(rendition.filter)
```

Comment:
For some reason this is not working as expected when the rendition file is removed from the filesystem.

Reply:
I guess I figured out what exactly is happening here. When we delete the rendition image file from filesystem (for eg: deleting from /bakerydemo/media/images in bakerydemo example) the file is deleted but the rendition model with that image's data is still stored somewhere. Moreover when the `get_rendition` method is called with same `filter` the rendition is not generated again (because the original image and filter are same) and therefore the rendition image is not reflected in images folder.

Reply:
One solution to this can be we use the code of `--purge` option by default. Therefore the command will first delete the existing rendition and then generate it.

Reply:
haha classic, we had the same idea, go for it.

Thanks for working through this, I know it is not as simple as we first thought.

This may also make your tests simpler so hopefully that other issue does not keep happening.

Reply:
😄 I will update the command and tests and push soon. Hoping to not break anything 😬